### PR TITLE
Add chain and channel monitors

### PR DESCRIPTION
**Pull request by TonyGiorgio**

### CODE REVIEW

File: node-manager/src/nodemanager.rs [hunk 1]
```diff
@@ -83,13 +88,15 @@ impl NodeManager {
...
-        let wallet = MutinyWallet::new(mnemonic.clone(), storage, Network::Testnet);
+        let wallet = MutinyWallet::new(mnemonic.clone(), storage.clone(), network);
 
         let node_storage = MutinyBrowserStorage::get_nodes().expect("could not retrieve node keys");
 
         NodeManager {
             mnemonic,
+            network,
             wallet,
+            storage: storage.clone(),
```
Comment (benthecarman) [inline node-manager/src/nodemanager.rs:99]: this .clone seems redundant

File: node-manager/src/wallet.rs [hunk 2]
```diff
@@ -178,3 +175,17 @@ fn get_tr_descriptors_for_extended_key(
...
+        Network::Bitcoin => Ok("https://blockstream.info/api"),
+        Network::Testnet => Ok("https://blockstream.info/testnet/api"),
+        Network::Signet => Ok("https://mempool.space/signet/api"),
+        Network::Regtest => Err(bdk::Error::Generic(
+            "No esplora client available for regtest".to_string(),
+        )),
+    }
+    .expect("What did I tell you about regtest?");
+    let blockchain = EsploraBlockchain::new(url, 20);
+    blockchain
```
Comment (benthecarman) [inline node-manager/src/wallet.rs:190]:
```suggestion
    EsploraBlockchain::new(url, 20)
```

**Comment by benthecarman** [comment]
pr title says channel monitor too but i dont see one

File: node-manager/src/node.rs [hunk 3]
```diff
@@ -30,23 +36,73 @@ pub(crate) type PeerManager = LdkPeerManager<
...
+        let chain_monitor: Arc<ChainMonitor> = Arc::new(chainmonitor::ChainMonitor::new(
+            None,
+            wallet.clone(),
+            Arc::new(MutinyLogger {}),
+            wallet,
+            persister.clone(),
+        ));
+
+        // read channelmonitor state from disk
+        let channelmonitors = persister
```
Comment (futurepaul) [inline node-manager/src/node.rs:97]: do we have any idea of how long this takes? sounds a little "blocking" to me. sure it's fine for now maybe just call it out as something we want potentially async / want this whole method to be async

File: node-manager/src/nodemanager.rs [hunk 4]
```diff
@@ -69,6 +71,9 @@ impl NodeManager {
     pub fn new(password: String, mnemonic: Option<String>) -> NodeManager {
```
Comment (futurepaul) [inline node-manager/src/nodemanager.rs:71]: I think it makes sense for this to return an `anyhow::Result<NodeManager>`, don't think it helps much to keep punting on it at this point

then we turn the expects into `?`

then we can do a try / catch in the browser

**Review by futurepaul — CHANGES_REQUESTED** [review]

File: node-manager/src/node.rs (hunk 3 above)
Comment (TonyGiorgio) [reply node-manager/src/node.rs:97]: It's just reading one key value, but yeah we'll end up needing async for this function eventually

File: node-manager/src/nodemanager.rs (hunk 1 above)
Comment (TonyGiorgio) [reply node-manager/src/nodemanager.rs]: fixed

File: node-manager/src/node.rs (hunk 3 above)
Comment (TonyGiorgio) [inline node-manager/src/node.rs:97]: @benthecarman this is where channel monitors is initialized, it's read in from the persister.

File: node-manager/src/nodemanager.rs (hunk 4 above)
Comment (TonyGiorgio) [reply node-manager/src/nodemanager.rs:71]: why `anyhow` ? Why not the mutiny errors?

Comment (TonyGiorgio) [reply node-manager/src/nodemanager.rs:71]: We can do it like this: https://rustwasm.github.io/wasm-bindgen/reference/types/result.html?highlight=result#resultt-e

```
impl Into<JsValue> for MutinyError {
    fn into(self) -> JsValue {
        todo!()
    }
}
```

Comment (TonyGiorgio) [reply node-manager/src/nodemanager.rs:71]: I don't know how to turn all the various errors here into `JsValue` right now. And I don't introduce any additional `expect`'s here so I'll open up a ticket and we can handle it soon outside of this PR.

Comment (TonyGiorgio) [reply node-manager/src/nodemanager.rs:71]: https://github.com/BitcoinDevShop/mutiny-web-poc/issues/38

**Comment by TonyGiorgio** [comment]
Handled what I could/should here, let me know if it's good to go.

**Review by benthecarman — APPROVED** [review]

**Comment by benthecarman** [comment]
needs rebase

**Pull request merged**